src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -604,8 +649,85 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
     auto config = selectedPd->getConfig();
     config.outConfs.front().setMemDesc(mem_desc);
-    //bypass any checks, we enforce the child descriptor
+    // bypass any checks, we enforce the child descriptor
     selectedPd->setConfig(config);
+
+    if (haveSubgraph()) {
+        // Adopt parent configuration, avoid to insert reorder before the MemoryInput.
+        std::vector<PortConfig> inConfs;
+        std::vector<Input::InputConfig> graphInputConfig;
+
+        auto mainInputDesc = getParentOutputMemDesc(getParentEdgeAt(0));
+        auto mainInputPrc = mainInputDesc->getPrecision();  // we have to align precision across all the inputs
+
+        inConfs.emplace_back(mainInputDesc);
+        graphInputConfig.emplace_back(node::Input::InputConfig{mainInputDesc, true});
+
+        for (size_t i = 1; i < getParentEdges().size(); i++) {
+            auto desc = getParentOutputMemDesc(getParentEdgeAt(i))->cloneWithNewPrecision(mainInputPrc);
+            inConfs.emplace_back(desc);
+            graphInputConfig.emplace_back(node::Input::InputConfig{desc, true});
+        }
+
+        std::vector<Input::OutputConfig> graphOutputConfig;
+        for (auto&& portConfig : config.outConfs) {
+            auto desc = portConfig.getMemDesc();
+            graphOutputConfig.emplace_back(node::Input::OutputConfig{desc, false});
+        }
+
+        // configure the inner graph to get the information about output memory descriptors
+        subGraph->Init(body, context, graphInputConfig, graphOutputConfig);
```

Comment:
This call must be inside `if(haveSubgraph())`.

Reply:
it is inside `if(haveSubgraph())`
https://github.com/xipingyan/openvino/blob/xp/whisper_readvalue_optimize/src/plugins/intel_cpu/src/nodes/memory.cpp#L655C1-L679C76